network/builder/src/builder.rs

```diff
@@ -153,53 +154,37 @@ impl NetworkBuilder {
             config.listen_address.clone(),
             authentication_mode,
         );
-        network_builder.add_connection_monitoring(
-            // TODO: Move these values into NetworkConfig
-            constants::PING_INTERVAL_MS,
-            constants::PING_TIMEOUT_MS,
-            constants::PING_FAILURES_TOLERATED,
-        );
-
-        // Sanity check seed peer addresses.
-        config
-            .verify_seed_peer_addrs()
-            .expect("Seed peer addresses must be well-formed");
-        let seed_peers = config.seed_peers.clone();
-
-        if config.mutual_authentication {
-            let network_peers = config.network_peers.clone();
-            let trusted_peers = if role == RoleType::Validator {
-                // for validators, trusted_peers is empty will be populated from consensus
-                HashMap::new()
-            } else {
-                network_peers
-            };
-
-            info!(
-                "network setup: role: {}, seed_peers: {:?}, trusted_peers: {:?}",
-                role, seed_peers, trusted_peers,
+        network_builder
+            .seed_addrs(config.seed_addrs.clone())
+            .seed_pubkeys(config.seed_pubkeys.clone())
+            .connectivity_check_interval_ms(config.connectivity_check_interval_ms)
+            .add_connection_monitoring(
+                // TODO: Move these values into NetworkConfig
+                constants::PING_INTERVAL_MS,
+                constants::PING_TIMEOUT_MS,
+                constants::PING_FAILURES_TOLERATED,
             );
 
-            network_builder
-                .trusted_peers(trusted_peers)
-                .seed_peers(seed_peers)
-                .connectivity_check_interval_ms(config.connectivity_check_interval_ms)
-                .add_connectivity_manager();
-        } else {
-            // Enforce the outgoing connection (dialer) verifies the identity of the listener (server)
-            if config.discovery_method == DiscoveryMethod::Onchain || !seed_peers.is_empty() {
-                network_builder
-                    .seed_peers(seed_peers)
-                    .add_connectivity_manager();
-            }
+        // Sanity check seed addresses.
+        config
+            .verify_seed_addrs()
+            .expect("Seed addresses must be well-formed");
+
+        // Don't turn on connectivity manager if we're a public facing vfn for example.
+        // TODO(philiphayes): could probably use a better way to specify these cases
+        if config.mutual_authentication
+            || config.discovery_method == DiscoveryMethod::Onchain
```

Comment:
curious why this is config.discovery != DiscoveryMethod::None -- but this is probably just as good...

Reply:
yeah this is a pretty weird condition imo, i just preserved the behaviour that was already there. thinking a bit more, i think the only cases where we don't want connmgr are server-only use cases. that this is hard to specify probably means we could be more explicit in our configs